scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1242,4 +1211,29 @@ class FormatOps(val tree: Tree, val initStyle: ScalafmtConfig) {
       case _ => (tree, select)
     }
 
+  @tailrec
+  final def findTokenWith[A](f: FormatToken => Option[A])(
+      ft: FormatToken,
+      step: Int,
+      end: Int = Int.MaxValue
```

Comment:
Maybe I missed it, but I don't found where `end` parameter passes into function. Looks like it always has default value

Reply:
will be used later. to look for something between `(` and `)`.

Reply:
what about this last patch? i removed end (the search can simply stop there) and changed step to `FormatToken`-hopping lambda.